### PR TITLE
[13762] Add custom stg templates as an option

### PR DESCRIPTION
**Pull request by ChrisFWP**
Extra option added to provide a template run time. Custom user code could be generated from idl files without recompiling the generator itself.

### CODE REVIEW

**Comment by MiguelCompany** [comment]
Hi @ChrisFWP
This looks like a nice addition indeed. Thank you for your contribution!
We will review and include it with the next release of Fast DDS Gen.

File: src/main/java/com/eprosima/fastdds/fastddsgen.java [hunk 1]
```diff
@@ -721,6 +760,30 @@ private Project parseIDL(
...
+                            else
+                            {
+                                project.addCommonSrcFile(ctx.getFilename() + entry.getValue());
+                            }
+                        }
+                    }
+                }
+            	
                 System.out.println("Generating Type definition files...");
                 if (returnedValue =
```
Comment (richiware) [inline src/main/java/com/eprosima/fastdds/fastddsgen.java:788]:
```suggestion
                if (returned && returnedValue =
```

**Review by richiware — CHANGES_REQUESTED** [review]

**Comment by richiware** [comment]
@ChrisFWP friendly ping

**Comment by MiguelCompany** [comment]
@richiware As this might come useful in the near future, I went ahead and applied your suggestions.

Would you mind reviewing again?

File: src/main/java/com/eprosima/fastdds/fastddsgen.java [hunk 2]
```diff
@@ -721,8 +768,36 @@ private Project parseIDL(
...
+                        else
+                        {
+                            break;
+                        }
+                    }
+                }
+            	
                 System.out.println("Generating Type definition files...");
-                if (returnedValue =
+                if (returned && returnedValue =
```
Comment (richiware) [inline src/main/java/com/eprosima/fastdds/fastddsgen.java:800]:
```suggestion
                if (returnedValue && returnedValue =
```

**Review by richiware — CHANGES_REQUESTED** [review]

File: src/main/java/com/eprosima/fastdds/fastddsgen.java (hunk 2 above)
Comment (MiguelCompany) [reply src/main/java/com/eprosima/fastdds/fastddsgen.java]: 0b58bdbe3b138bbb10dbbb3a70d831e0b1e4211f

**Review by richiware — APPROVED** [review]

**Comment by MiguelCompany** [comment]
@mergifyio rebase

**Review by richiware — APPROVED** [review]

**Pull request merged**